### PR TITLE
Add profiler and geth tests to metacoin

### PR DESCRIPTION
**Pull request by LogvinovLeon**
## Description

This PR adds a README for metacoin project.
It also adds profiling capabilities to it and tests running against geth.

## Motivation and Context

To have a small, well-documented example.

## How Has This Been Tested?

Manually & tests.

## Types of changes

*   [ ] Bug fix (non-breaking change which fixes an issue)
*   [x] New feature (non-breaking change which adds functionality)
*   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [x] Change requires a change to the documentation.
*   [x] Added tests to cover my changes.
*   [x] Added new entries to the relevant CHANGELOG.jsons.
*   [x] Labeled this PR with the 'WIP' label if it is a work in progress.
*   [x] Labeled this PR with the labels corresponding to the changed package.

### CODE REVIEW

File: packages/metacoin/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+## @0xproject/metacoin
+
+This is a small example project that server as a showroom/polygon for 0x dev tools.
+It suppports:
+
+*   Compiling & testing smart contracts
+*   Typed contract wrappers
```
Comment (fabioberger) [inline packages/metacoin/README.md]: `Generating typed contract wrappers`

File: packages/metacoin/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+*   Coverage
```
Comment (fabioberger) [inline packages/metacoin/README.md]: Solidity coverage

File: packages/metacoin/README.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+*   Profiling
```
Comment (fabioberger) [inline packages/metacoin/README.md]: Solidity gas profiling

File: packages/metacoin/README.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+*   Running tests against Ganache
+*   Running tests against our fork of Geth that supports snapshots & time travel
```
Comment (fabioberger) [inline packages/metacoin/README.md]: Running tests against our fork of Geth (it supports snapshotting & time travel)

File: packages/metacoin/README.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+### Coverage
+
+```bash
+yarn test:coverage
+yarn coverage:report:html
+```
+
+### Profiling
+
+Please note that traces emmited by ganache have incorrect gas costs so we recommend using Geth for profiling.
```
Comment (fabioberger) [inline packages/metacoin/README.md]: Typo: `emitted`

File: packages/metacoin/test/global_hooks.ts [hunk 6]
```diff
@@ -1,10 +1,15 @@
 import { env, EnvVars } from '@0xproject/dev-utils';
 
 import { coverage } from './utils/coverage';
+import { profiler } from './utils/profiler';
 
-after('generate coverage report', async () => {
+after('generate coverage & profiler report', async () => {
```
Comment (fabioberger) [inline packages/metacoin/test/global_hooks.ts]: `||` not `&` right?

File: packages/metacoin/test/metacoin_test.ts [hunk 7]
```diff
@@ -10,6 +10,7 @@ import { MetacoinContract, TransferContractEventArgs } from '../src/contract_wra
 
 import { chaiSetup } from './utils/chai_setup';
 import { config } from './utils/config';
+// import { profiler } from './utils/profiler';
```
Comment (fabioberger) [inline packages/metacoin/test/metacoin_test.ts]: Add comment above this and those below:

```
// Comment out the next line enable profiling
```

File: packages/metacoin/test/utils/web3_wrapper.ts [hunk 8]
```diff
@@ -1,31 +1,78 @@
...
+    Ganache = 'ganache',
+    Geth = 'geth',
+}
+
+let testProvider: ProviderType;
+switch (process.env.TEST_PROVIDER) {
+    case undefined:
+        testProvider = ProviderType.Ganache;
+        break;
+    case 'ganache':
```
Comment (fabioberger) [inline packages/metacoin/test/utils/web3_wrapper.ts]: You can merge these cases:

```
case undefined:
case 'ganache':
     testProvider = ProviderType.Ganache;
     break;
```

File: packages/metacoin/test/utils/web3_wrapper.ts [hunk 9]
```diff
@@ -1,31 +1,78 @@
...
+        testProvider = ProviderType.Ganache;
+        break;
+    case 'geth':
+        testProvider = ProviderType.Geth;
+        break;
+    default:
+        throw new Error(`Unknown TEST_PROVIDER: ${process.env.TEST_PROVIDER}`);
```
Comment (fabioberger) [inline packages/metacoin/test/utils/web3_wrapper.ts]: `errorUtils.spawnSwitchErr()`

File: packages/metacoin/test/utils/web3_wrapper.ts (hunk 8 above)
Comment (LogvinovLeon) [reply packages/metacoin/test/utils/web3_wrapper.ts]: no-fallthrough-switch-cases

**Pull request merged**